### PR TITLE
Drop support for EOLed Kubernetes versions

### PR DESCRIPTION
**Pull request by michi-covalent**
Remove EOLed Kubernetes versions from Ginkgo PR and scheduled runs, and update the docs accordingly.

Ref: https://kubernetes.io/releases/patch-releases/

### CODE REVIEW

**Review by sayboras — APPROVED** [review]
LGTM ✅

**Review by ghost — APPROVED** [review]

**Comment by michi-covalent** [comment]
i'll bring this up in the community meeting today

**Review by aanm — CHANGES_REQUESTED** [review]
To accept the PR we should do a couple more things:
* Document our testing criteria in the documentation, under Documentation/network/kubernetes/compatibility.rst and Documentation/network/kubernetes/requirements.rst should be fine
* Keep the list of cloud provider k8s versions that we test in the list on those files.
* Making sure that by removing the k8s versions that we aren't dropping testing on specific kernel versions. For example, kernel 4.19 is only tested with k8s 1.19 so if we are dropping it then we need to change the test of 4.19 to be 1.26. That's defined in .github/actions/ginkgo/main-k8s-versions.yaml.

**Review by viktor-kurchenko — APPROVED** [review]

**Comment by michi-covalent** [comment]
thanks for the feedback andre, i'll update the docs.

regarding the kernel version, don't we also test 4.19 with k8s 1.27? https://github.com/cilium/cilium/blob/main/.github/actions/ginkgo/main-k8s-versions.yaml#L11-L16

**Comment by michi-covalent** [comment]
back to draft to address feedback

File: Documentation/network/kubernetes/compatibility.rst [hunk 1]
```diff
@@ -18,12 +18,30 @@ with Cilium. Older Kubernetes versions not listed in this table do not have
...
+
+As a general rule, Cilium aims to run e2e tests using the latest build from the
+development branch against currently supported Kubernetes versions defined in
+`Kubernetes Patch Releases <https://kubernetes.io/releases/patch-releases/>`_
+page.
+
+Once the release branch gets created from the development branch, Kubernetes
+versions Cilium runs e2e tests against for that release branch get finalized.
+Cilium typically does not change the Kubernetes versions it uses to run e2e
+tests for the entire support period of the release.
```
Comment (michi-covalent) [inline Documentation/network/kubernetes/compatibility.rst]:
```suggestion
tests for the entire maintenance period of the release.
```

File: Documentation/network/kubernetes/compatibility.rst [hunk 2]
```diff
@@ -18,12 +18,30 @@ with Cilium. Older Kubernetes versions not listed in this table do not have
 Cilium support. Newer Kubernetes versions, while not listed, will depend on the
 backward compatibility offered by Kubernetes.
```
Comment (michi-covalent) [inline Documentation/network/kubernetes/compatibility.rst]: maybe something like:

Older and newer Kubernetes versions, while not listed, will depend on the
forward / backward compatibility offered by Kubernetes.

Comment (aanm) [reply Documentation/network/kubernetes/compatibility.rst]: LGTM

File: Documentation/network/kubernetes/compatibility.rst [hunk 3]
```diff
@@ -18,12 +18,30 @@ with Cilium. Older Kubernetes versions not listed in this table do not have
...
+Cilium typically does not change the Kubernetes versions it uses to run e2e
+tests for the entire support period of the release.
+
+Additionally, Cilium runs e2e tests against various cloud providers' managed
+Kubernetes offerings using multiple Kubernetes versions. See the following links
+for the current test matrix for each cloud provider:
+
+- :git-tree:`AKS <.github/actions/azure/k8s-versions.yaml>`
+- :git-tree:`EKS <.github/actions/aws/k8s-versions.yaml>`
+- :git-tree:`GKE <.github/actions/gke/k8s-versions.yaml>`
```
Comment (aanm) [inline Documentation/network/kubernetes/compatibility.rst:42]: Nice trick :clap:

**Review by aanm — CHANGES_REQUESTED** [review]
LGTM but there's one bullet point missing:
> Making sure that by removing the k8s versions that we aren't dropping testing on specific kernel versions. For example, kernel 4.19 is only tested with k8s 1.19 so if we are dropping it then we need to change the test of 4.19 to be 1.26. That's defined in .github/actions/ginkgo/main-k8s-versions.yaml.

**Comment by michi-covalent** [comment]
> Making sure that by removing the k8s versions that we aren't dropping testing on specific kernel versions. For example, kernel 4.19 is only tested with k8s 1.19 so if we are dropping it then we need to change the test of 4.19 to be 1.26. That's defined in .github/actions/ginkgo/main-k8s-versions.yaml.

thanks @aanm, yeah this is the part i wasn't sure about. don't we already test 4.19 with 1.27, or am i reading this wrong? 💭  https://github.com/cilium/cilium/blob/main/.github/actions/ginkgo/main-k8s-versions.yaml#L11-L16

**Comment by michi-covalent** [comment]
requesting additional review from @joestringer 🚀 🙏

**Review by aanm — APPROVED** [review]

**Pull request merged**